lib/modules/datasource/hex/schema.ts

```diff
@@ -0,0 +1,48 @@
+import { z } from 'zod';
+import { LooseArray } from '../../../util/schema-utils';
+import type { Release, ReleaseResult } from '../types';
+
+export const HexRelease = z
+  .object({
+    html_url: z.string().optional(),
+    meta: z
+      .object({
+        links: z.object({
+          Github: z.string(),
+        }),
+      })
+      .transform((meta) => meta.links.Github)
+      .nullable()
+      .catch(null),
+    releases: LooseArray(
+      z
+        .object({
+          version: z.string(),
+          inserted_at: z.string().optional(),
+        })
+        .transform(({ version, inserted_at: releaseTimestamp }): Release => {
+          const release: Release = { version };
+
+          if (releaseTimestamp) {
+            release.releaseTimestamp = releaseTimestamp;
+          }
+
+          return release;
+        })
+    ).refine((releases) => releases.length > 0, 'No releases found'),
+  })
+  .transform(
```

Comment:
I find it a little strange how you use transform in both the first part of this parsing as well as here - there doesn't seem to be a clear separation between validating/parsing the data and then transforming it